pkg/sql/sem/builtins/builtins.go

```diff
@@ -8952,6 +8952,75 @@ WHERE object_id = table_descriptor_id
 			Volatility: volatility.Stable,
 		},
 	),
+	"crdb_internal.backup_compaction": makeBuiltin(
+		tree.FunctionProperties{Undocumented: true},
+		tree.Overload{
+			Types: tree.ParamTypes{
+				{Name: "collection_uri", Typ: types.StringArray},
+				{Name: "full_backup_path", Typ: types.String},
+				{Name: "start_time", Typ: types.TimestampTZ},
+				{Name: "end_time", Typ: types.TimestampTZ},
+			},
+			ReturnType: tree.FixedReturnType(types.Void),
+			Fn: func(ctx context.Context, evalCtx *eval.Context, args tree.Datums) (tree.Datum, error) {
+				manager, err := evalCtx.BackupCompactionManagerFactory.GetBackupCompactionManager(ctx)
+				if err != nil {
+					return nil, err
+				}
+				ary := *tree.MustBeDArray(args[0])
+				collectionURI, ok := darrayToStringSlice(ary)
+				if !ok {
+					return nil, errors.Newf("expected array value, got %T", args[0])
+				}
+				encryption := jobspb.BackupEncryptionOptions{
+					Mode: jobspb.EncryptionMode_None,
+				}
+				fullPath := string(tree.MustBeDString(args[1]))
+				start, end := tree.MustBeDTimestampTZ(args[2]), tree.MustBeDTimestampTZ(args[3])
+				startTs := hlc.Timestamp{WallTime: start.UnixNano()}
+				endTs := hlc.Timestamp{WallTime: end.UnixNano()}
+				return tree.DVoidDatum, manager.CompactBackups(
+					ctx, collectionURI, nil, fullPath, encryption, startTs, endTs,
```

Comment:
Right now, I just have a `nil` incremental location, which means that it defaults to the default incremental location.

To cover cases where the customer has set a different `incremental_location`, I was going to add an overload where an `incremental_location` can be provided. This does mean two more overloads are needed, one where encryption is provided and one where it isn't.

However, I do think this is starting to get excessive in terms of the number of overloads required. I'm thinking I can make `encryption` a required argument, and to use the builtin, you would just have to `crdb_internal.json_to_pb('jobspb.BackupEncryptionOption', '{"mode": 2}')`, but the calling of the builtin is getting quite verbose.

Any thoughts on how this could be improved? @msbutler @dt

Reply:
I think it's worth building this  out incrementally: I would have first landed the simplest built-in (without encryption or a different incremental location support). But since you already have support for encryption, how about we land one incantation, without any overloads, for now: 
```
{Name: "collection_uri", Typ: types.StringArray},
{Name: "full_backup_path", Typ: types.String},
{Name: "start_time", Typ: types.TimestampTZ},
{Name: "end_time", Typ: types.TimestampTZ},
{Name: "encryption_opts", Typ: types.Bytes}, // Last arg
```

If the encryption arg is an empty byte array, then we can assume the caller doesn't need encryption. Recall that crdb_internal functions are not going to be used for customers, just us-- so the builtin doesn't need to be pretty!

Reply:
Yeah that sounds good, I'll leave the incremental location addition to a different PR then.